File: 1492-the-kth-factor-of-n/main.go
func kthFactor(n int, k int) int {
    
    var factorList []int
    for i := 1; i <= n; i++ {
        if n % i == 0 {
            factorList = append(factorList, i)
        }
    }

    if k - 1 >= len(factorList) { return - 1}

    return factorList[k - 1]
}

File: 1492-the-kth-factor-of-n/main.py
class Solution:
    def kthFactor(self, n: int, k: int) -> int:
        
        factor_list = [idx for idx in range(1, n + 1) if n % idx == 0]

        if k - 1 >= len(factor_list):
            return -1

        return factor_list[k - 1]
